scripts/generate-compatibility-json.py

```diff
@@ -0,0 +1,343 @@
+SKIP = "skip"
+# fields will contain arbitrary key value pairs in comments that use the compatibility flag.
+FIELDS = "fields"
+TEST_SUITE = "test_suite"
+TESTS = "tests"
+RELEASES_URL = "https://api.github.com/repos/cosmos/ibc-go/releases"
+CHAIN_A = "chain-a"
+CHAIN_B = "chain-b"
+HERMES = "hermes"
+DEFAULT_IMAGE = "ghcr.io/cosmos/ibc-go-simd"
+RLY = "rly"
+# MAX_VERSION is a version string that will be greater than any other semver version.
+MAX_VERSION = "9999.999.999"
+
+
+def parse_version(version: str) -> semver.Version:
+    """
+    parse_version takes in a version string which can be in multiple formats,
+    and converts it into a valid semver.Version which can be compared with each other.
+    The version string is a docker tag. It can be in the format of
+    - main
+    - v1.2.3
+    - 1.2.3
+    - release-v1.2.3 (a tagged release)
+    - release-v1.2.x (a release branch)
+    """
+    if version.startswith("v"):
+        # semver versions do not include a "v" prefix.
+        version = version[1:]
+    if version.startswith("release-"):
+        # strip off the release prefix and parse the actual version
+        return parse_version(version[len("release-"):])
+    # ensure "main" is always greater than other versions for semver comparison.
+    if version == "main":
+        # main will always be the newest release.
+        version = MAX_VERSION
+    if version.endswith("x"):
+        # we always assume the release branch is newer than the previous release.
+        # for example, release-v9.0.x is newer than release-v9.0.1
+        version = version.replace("x", "999", 1)
+    return semver.Version.parse(version)
+
+
+def parse_args() -> argparse.Namespace:
+    """Parse command line arguments."""
+    parser = argparse.ArgumentParser(description="Generate Compatibility JSON.")
+    parser.add_argument(
+        "--file",
+        help="The test file to look at.",
+    )
+    parser.add_argument(
+        "--release-version",
+        help="The version to run tests for.",
+    )
+    parser.add_argument(
+        "--image",
+        default=DEFAULT_IMAGE,
+        help=f"Specify the image to be used in the test. Default: {DEFAULT_IMAGE}",
+    )
+    parser.add_argument(
+        "--relayer",
+        choices=[HERMES, RLY],
+        default=HERMES,
+        help=f"Specify relayer, either {HERMES} or {RLY}",
+    )
+    parser.add_argument(
+        "--chain",
+        choices=["chain-a", "chain-b", "all"],
+        default="all",
+        help=f"Specify the chain to run tests for (chain-a, chain-b, all)",
+    )
+    return parser.parse_args()
+
+
+def main():
+    args = parse_args()
+
+    file_metadata = _build_file_metadata(args.file)
+    tags = _get_ibc_go_releases(args.release_version)
+
+    # extract the "from_version" annotation specified in the test file.
+    # this will be the default minimum version that tests will use.
+    min_version = parse_version(file_metadata[FIELDS][FROM_VERSION])
```

Comment:
hm, do we want to allow this to be unset? Basically if a new test is added and versioning info is not added it will attempt to run it against all versions `<= args.release_version` no? I guess this is an easy error to spot but might make more sense to enforce all tests have the annotation?

Reply:
e.g generated these for `client_test.go` and see `TestAllowedClientsParam` gen's a run for all pairs versions.